luma/core/bitmap_font.py

```diff
@@ -0,0 +1,632 @@
+        sprite_table.close()
+        return self
+
+    def load_sprite_table(self, sprite_table, index, xwidth, glyph_size, cell_size, mappings=None):
+        """
+        Load a font from a sprite table
+
+        :param sprite_table: A PIL.Image representation of every glyph within the font
+        :type sprite_table: PIL.Image
+        :param index: The list of character values contained within sprite_table.
+            This list MUST be in the same order that the glyphs for the characters
+            appear within the sprite_table (in left to right, top to bottom order)
+        :type index: list or other iterable
+        :param xwidth: number of pixels between placements of each character in a
+            line of text
+        :type xwidth: int
+        :param glyph_size: tuple containing the width and height of each character
+            in the font
+        :type glyph_size: tuple(int, int)
+        :param cell_size: tuple containing the width and height of each cell in the
+            sprite table.  Defaults to the size of the glyphs.
+        :type cell_size: tuple(int, int)
+        :param mappings: a dictionary of unicode to value pairs (optional)
+            Mappings allow the appropriate unicode values to be provided for
+            each character contained within the font
+        :type mappings: dict
+
+        .. note:
+            Font contained within table must adhere to the following conditions
+            * All glyphs must be the same size
+            * Glyphs are contained within the sprite table in a grid arrangement
+            * The grid is filled with glyphs placed in horizontal order
+            * Each cell in the grid is the same size
+            * The placement of each glyph has no offset from its origin
+        """
+
+        table_width = sprite_table.size[0]
+
+        # Each character uses the same data
+        line = [xwidth, 0, 0, -glyph_size[1], glyph_size[0], 0]
+
+        data = []
+        # Generate an entry for each character
+        for c in range(len(index)):
+            offset = c * cell_size[0]
+            left = offset % table_width
+            top = (offset // table_width) * cell_size[1]
+            right = left + glyph_size[0]
+            bottom = top + glyph_size[1]
+            data = data + line + [left, top, right, bottom]
+        self._populate_metrics(sprite_table, self._from_16_to_8(data), index, mappings)
+        return self
+
+    def save(self, filename):
+        """
+        Write luma.core.bitmap_font data to a file
+        """
+        with open(filename, 'wb') as fp:
+            fontdata = self._generate_fontdata()
+            fp.write(b'LUMA.CORE.BITMAP_FONT\n')
+            cbor2.dump(fontdata, fp)
+
+    def dumps(self):
+        """
+        Return luma.core.bitmap_font data serialized as a string
+        """
+        fontdata = self._generate_fontdata()
+        return cbor2.dumps(fontdata)
+
+    def _generate_fontdata(self):
```

Comment:
this can probably be a standalone utility method instead of using it in this class only. could be done in other pr as well.

Reply:
I wasn't clear on what you were suggesting.  _generate_fontdata is only relevant to bitmap_font.